os/drivers/lcd/mipi_lcd.c

```diff
@@ -48,34 +53,160 @@ struct mipi_lcd_dev_s {
 	int fb_alloc_count;
 	uint8_t power;
 	struct mipi_lcd_config_s *config;
+	LCD_MODE mode;
 };
 
 static struct mipi_lcd_dev_s g_lcdcdev;
 
-static int send_cmd(struct mipi_lcd_dev_s *priv, lcm_setting_table_t command)
+static int switch_to_cmd(struct mipi_lcd_dev_s *priv)
 {
 	int transfer_status = OK;
-	u8 cmd = command.cmd;
-	u8 *cmd_addr = command.para_list;
-	u32 payload_len = command.count;
+	priv->config->lcd_mode_switch(false);
+	uint8_t cmd_mode[2] = {0x00, 0x00};
 	struct mipi_dsi_msg msg;
-	msg.channel = cmd;
-	if (payload_len == 0) {
-		msg.type = MIPI_DSI_DCS_SHORT_WRITE_0_PARAM;
-	} else if (payload_len == 1) {
-		msg.type = MIPI_DSI_DCS_SHORT_WRITE_1_PARAM;
-	} else {
-		msg.type = MIPI_DSI_DCS_LONG_WRITE;
-	}
-	msg.tx_buf = cmd_addr;
-	msg.tx_len = payload_len;
+	msg.channel = 0xB0;
+	msg.type = MIPI_DSI_DCS_LONG_WRITE;
+	msg.tx_buf = cmd_mode;
+	msg.tx_len = 2;
 	msg.flags = 0;
+	transfer_status = mipi_dsi_transfer(priv->dsi_dev, &msg);
+	if (transfer_status != OK) {
+		lcddbg("Command %x not sent \n", msg.channel);
+	}
+	priv->mode = COMMAND;
+	return transfer_status;
+}
+
+static int switch_to_video(struct mipi_lcd_dev_s *priv)
+{
+	priv->mode = VIDEO;
+	int transfer_status = OK;
 	priv->config->lcd_mode_switch(false);
+	uint8_t cmd_mode[2] = {0x00, 0x10};
+	struct mipi_dsi_msg msg;
+	msg.channel = 0xB0;
+	msg.type = MIPI_DSI_DCS_LONG_WRITE;
+	msg.tx_buf = cmd_mode;
+	msg.tx_len = 2;
+	msg.flags = 0;
 	transfer_status = mipi_dsi_transfer(priv->dsi_dev, &msg);
+	if (transfer_status != OK) {
+		lcddbg("Command %x not sent \n", msg.channel);
+	}
 	priv->config->lcd_mode_switch(true);
+	priv->mode = VIDEO;
+	return transfer_status;
+}
+
+static int send_cmd(struct mipi_lcd_dev_s *priv, bool display_on)
+{
+	priv->config->lcd_mode_switch(false);
+	int transfer_status = OK;
+	struct mipi_dsi_msg msg;
+	if (display_on) {
+		msg.channel = 0x29;
+	} else {
+		msg.channel = 0x28;
+	}
+	msg.type = MIPI_DSI_DCS_LONG_WRITE;
+	msg.tx_len = 0;
+	msg.flags = 0;
+	transfer_status = mipi_dsi_transfer(priv->dsi_dev, &msg);
 	if (transfer_status != OK) {
 		lcddbg("Command %x not sent \n", cmd);
 	}
+	if (priv->mode == VIDEO) {
+		priv->config->lcd_mode_switch(true);
+	}
+	return transfer_status;
+}
+
+static int send_lcd_init_data(struct mipi_lcd_dev_s *priv)
+{
+	int transfer_status = OK;
+	struct mipi_dsi_msg msg_data, msg;
+	uint32_t size = CONFIG_AMEBASMART_MIPI_PAYLOAD_LEN;
+	fb_coord_t curr_row_start = 0;
+	fb_coord_t curr_row_end;
+	fb_coord_t curr_col_start = 0;
+	fb_coord_t curr_col_end = LCD_XRES - 1;
+	fb_coord_t width_size = LCD_XRES * 2;
+	uint8_t *init_buffer = (uint8_t *)kmm_malloc(CONFIG_AMEBASMART_MIPI_PAYLOAD_LEN);
```

Comment:
Hello @anjana348, in TizenLite driver structure, we cannot apply chipset specific config directly. Thus, I think some alternative method should be applied.
For example, adding another variable such as `u32 mipi_buffer_max_size;` in the `struct mipi_lcd_dev_s`. And then during initialization, the chipset specific code can pass in the variable and store in the struct.
eg: `priv-> mipi_buffer_max_size = variable passed in from chipset BSP`
So, all the condition check can rely on the priv structure instead of chipset specific.